lib/spack/spack/util/executable.py

```diff
@@ -81,75 +88,54 @@ def path(self):
 
-        Arguments
-          args
-            command line arguments to the executable to run.
-
-        Optional arguments
-
-          fail_on_error
-
-            Raise an exception if the subprocess returns an
-            error. Default is True.  When not set, the return code is
-            available as `exe.returncode`.
-
-          ignore_errors
-
-            An optional list/tuple of error codes that can be
-            *ignored*.  i.e., if these codes are returned, this will
-            not raise an exception when `fail_on_error` is `True`.
-
-          output, error
-
-            These arguments allow you to specify new stdout and stderr
-            values.  They default to `None`, which means the
-            subprocess will inherit the parent's file descriptors.
-
-            You can set these to:
-            - python streams, e.g. open Python file objects, or os.devnull;
-            - filenames, which will be automatically opened for writing; or
-            - `str`, as in the Python string type. If you set these to `str`,
-               output and error will be written to pipes and returned as
-               a string.  If both `output` and `error` are set to `str`,
-               then one string is returned containing output concatenated
-               with error.
-
-          input
-
-            Same as output, error, but `str` is not an allowed value.
-
-        Deprecated arguments
-
-          return_output[=False]
-
-            Setting this to True is the same as setting output=str.
-            This argument may be removed in future Spack versions.
-
+        Parameters:
+            *args (str): Command-line arguments to the executable to run
+
+        Keyword Arguments:
+            env (dict): The environment to run the executable with
+            fail_on_error (bool): Raise an exception if the subprocess returns
+                an error. Default is True. The return code is available as
+                ``exe.returncode``
+            ignore_errors (int or list): A list of error codes to ignore.
+                If these codes are returned, this process will not raise
+                an exception even if ``fail_on_error`` is set to ``True``
+            input: Where to read stdin from
+            output: Where to send stdout
+            error: Where to send stderr
+
+        Accepted values for input, output, and error:
+
+        * python streams, e.g. open Python file objects, or ``os.devnull``
+        * filenames, which will be automatically opened for writing
+        * ``str``, as in the Python string type. If you set these to ``str``,
+          output and error will be written to pipes and returned as a string.
+          If both ``output`` and ``error`` are set to ``str``, then one string
+          is returned containing output concatenated with error. Not valid
+          for ``input``
+
+        By default, the subprocess inherits the parent's file descriptors.
```

Comment:
This isn't actually making it into the documentation because it's a double-underscore method. I think Sphinx has an option to add these methods to the documentation. @alalazo?